lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -2284,9 +2291,15 @@ namespace {
     /// context and false otherwise.
     bool isDistributedAccess(SourceLoc declLoc, ValueDecl *decl,
                              Expr *context) {
+      fprintf(stderr, "[%s:%d] (%s) decl:\n", __FILE__, __LINE__, __FUNCTION__);
+      decl->dump();
+
       // If base of the call is 'local' we permit skip distributed checks.
-      if (auto baseSelf = findReferencedBaseSelf(context)) {
-        if (baseSelf->getAttrs().hasAttribute<KnownToBeLocalAttr>())
+      if (auto baseSelf = findReferencedCallBase(context)) {
+        fprintf(stderr, "[%s:%d] (%s) base:\n", __FILE__, __LINE__, __FUNCTION__);
+        baseSelf->dump();
+
+        if (baseSelf->isDistributedKnownToBeLocal())
```

Comment:
Oh, so this isn't necessarily `self`, but `isDistributedKnownToBeLocal` will be true for `self`. Okay.

Reply:
Right, self is a special case that is known local :)